src/core/control/tools/InputHandler.cpp

```diff
@@ -16,47 +16,44 @@
 
 #include "filesystem.h"  // for path
 
-InputHandler::InputHandler(XournalView* xournal, XojPageView* redrawable, const PageRef& page):
-        xournal(xournal), redrawable(redrawable), page(page), stroke(nullptr) {}
+InputHandler::InputHandler(XournalView* xournal, const PageRef& page): xournal(xournal), page(page) {}
 
 InputHandler::~InputHandler() = default;
 
-/**
- * @return Current editing stroke
- */
-auto InputHandler::getStroke() -> Stroke* { return stroke; }
+auto InputHandler::getStroke() const -> Stroke* { return stroke.get(); }
```

Comment:
It looks like, that stroke will never be initialized anymore, how about removing this then?